File: app/src/main/java/com/practice/forecast/ui/map/MapFragmentModule.java
package com.practice.forecast.ui.map;

import com.practice.forecast.App;
import com.practice.forecast.AppComponent;
import com.practice.weathermodel.logger.Logger;

import androidx.lifecycle.ViewModelProvider;
import dagger.Module;
import dagger.Provides;

@Module
public class MapFragmentModule {

    private final AppComponent appComponent;

    public MapFragmentModule() {
        this.appComponent = App.getInstance().getAppComponent();
        appComponent.injectMapFragment(this);
    }

    @Provides
    ViewModelProvider.Factory provideWeatherViewModelFactory(){
        return new WeatherViewModelFactory(new WeatherMapViewModel(appComponent.provideNetworkClient(),
                appComponent.provideLocationSupplier(), Logger.withTag("MyLog"),
                appComponent.providePrefs(),
                appComponent.provideUtils(),
                appComponent.provideNetworkReceiver()));
    }
}

File: app/src/main/java/com/practice/forecast/ui/map/MapFragmentModule.kt
package com.practice.forecast.ui.map

import androidx.lifecycle.ViewModelProvider
import com.practice.forecast.AppComponent
import com.practice.weathermodel.logger.Logger
import dagger.Module
import dagger.Provides

@Module
class MapFragmentModule {
    private val appComponent: AppComponent? = com.practice.forecast.App.instance?.appComponent

    @Provides
    fun provideWeatherViewModelFactory(): ViewModelProvider.Factory {
        return WeatherViewModelFactory(WeatherMapViewModel(appComponent!!.provideNetworkClient(),
                appComponent.provideLocationSupplier(), Logger.withTag("MyLog"),
                appComponent.providePrefs(),
                appComponent.provideUtils(),
                appComponent.provideNetworkReceiver()))
    }

    init {
        appComponent?.injectMapFragment(this)
    }
}